ext/sysvshm/sysvshm.c

```diff
@@ -345,6 +409,10 @@ static zend_long php_check_shm_data(sysvshm_chunk_head *ptr, zend_long key)
 	zend_long pos;
 	sysvshm_chunk *shm_var;
 
+	if (!ptr) {
```

Comment:
These checks should be redundant now, as you check for null earlier than that, right?

Reply:
Yes, that's true. I left this there so that `php_check_shm_data` is not accidentally called without the prior checks.

But yes, we are probably better off not doing this here and thus require an explicit check beforehand.

Reply:
It might make sense then to turn the check into an assertion (`ZEND_ASSERT()`).

Reply:
Sounds good! Before I commit the assertion, what is your stance about the class name? Should we go with the current one or the new ones are better?

Reply:
IMO `SysvSharedMemory` is fine. Adding `Block` at the end doesn't seem to add useful information, and not prefixing (with `Sysv`) could be confusing wrt. ext/shmop.

Reply:
OK, thanks!